packages/notifier/src/publish-kit.js

```diff
@@ -15,6 +15,8 @@ const makeQuietRejection = reason => {
   void E.when(rejection, sink, sink);
   return rejection;
 };
+const makeTooFarRejection = () =>
+  makeQuietRejection(new Error('Cannot read past end of iteration.'));
```

Comment:
So I actually found one more leak, at least in v8. Apparently the `Error`s structured stack keeps references to each call site, including in this case the context object of the `finish` call. That object is the `{state, facets}` context of the notifier exo kit. While this is most likely not a problem in XS, and likely applies to other Exo class based patterns, I figured it'd be easy enough to fix in this case since I don't believe there is much value in capturing the stack of when the publisher was finished.

![image](https://user-images.githubusercontent.com/86499/233796122-6edfc24e-66cc-4788-b354-0f5ad6ae022f.png)


```suggestion
const tooFarRejection = harden(
  makeQuietRejection(new Error('Cannot read past end of iteration.')),
);
```

Reply:
Fixed.

Reply:
So the reason I suggested to `harden` was because the rejection would be shared instead of built for each publisher when reaching a terminal state.

Reply:
Yep, makes sense. And `makeQuietRejection` already hardens the returned promise.